src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelector.java

```diff
@@ -140,10 +140,28 @@ private static boolean isInstancePrimitive(@Nonnull Class<?> type, @Nonnull Obje
         return findMatchingMethod(receiver, method, args);
     }
 
+    private static boolean isEnumInitializer(Method m) {
+        if (!m.getDeclaringClass().isEnum()) {
+            return false;
+        }
+        if (!m.getName().equals("$INIT")) {
```

Comment:
It appears it isn't. How about checking that the method is synthetic instead? No other method on an enum type should take `Object[]` as arguments. That said, I don't think the JLS guarantees that the argument types will be `Object[]`, so it's still not perfect, only better.